src/runner/main.cpp

```diff
@@ -34,6 +46,49 @@ void chdir_current_executable()
     }
 }
 
+wil::unique_mutex_nothrow create_runner_mutex(const bool acquire, const bool msix_version)
+{
+    wchar_t username[UNLEN + 1];
+    DWORD username_length = UNLEN + 1;
+    GetUserNameW(username, &username_length);
+
+    return wil::unique_mutex_nothrow{ CreateMutexW(nullptr, acquire, (std::wstring(msix_version ? MSIX_VERSION_MUTEX_NAME : MSI_VERSION_MUTEX_NAME) + username).c_str()) };
+}
+
+void notify_if_msi_version_is_running()
+{
+    if (!create_runner_mutex(true, false) || GetLastError() == ERROR_ALREADY_EXISTS)
+    {
+        notifications::show_toast(localized_strings::MSI_VERSION_IS_ALSO_RUNNING);
+    }
+}
+
+void start_msi_uninstallation_sequence()
+{
+    const auto package_path = get_msi_package_path();
+
+    if (package_path.empty())
+    {
+        // No MSI version detected
+        return;
+    }
+
+    if (!offer_msi_uninstallation())
+    {
+        // User declined to uninstall

```

Comment:
In this case, if the MSI version is also running, we show the Error message and exit.